tests/performance-grpc/steps/steps.go

```diff
@@ -0,0 +1,123 @@
+package steps
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/kyma-project/istio/operator/tests/testcontext"
+	"istio.io/istio/pkg/test/util/tmpl"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"os"
+	"sigs.k8s.io/yaml"
+)
+
+const loadTestingNamespace = "load-testing"
+
+type TemplatedPerformanceJob struct {
+	templatedValues map[string]string
+	jobName         string
+}
+
+func (t *TemplatedPerformanceJob) SetTemplateValue(key, value string) error {
+	if t.templatedValues == nil {
+		t.templatedValues = make(map[string]string)
+	}
+
+	t.templatedValues[key] = value
+	return nil
+}
+
+func getRequestHost() string {
+	kymaDomain := os.Getenv("KYMA_DOMAIN")
+	if kymaDomain == "" {
+		print("KYMA_DOMAIN env variable is not set, using default address set in the Job template")
+		return ""
+	}
+
+	return fmt.Sprintf("%s.%s:443", "hello", kymaDomain)
+}
+
+func getRequestAuthority() string {
+	kymaDomain := os.Getenv("KYMA_DOMAIN")
+	if kymaDomain == "" {
+		print("KYMA_DOMAIN env variable is not set, using default address set in the Job template")
+		return ""
+	}
+
+	return fmt.Sprintf("%s.%s", "hello", kymaDomain)
+}
+
+//go:embed job.yaml
+var jobTemplate string
+
+func (t *TemplatedPerformanceJob) ExecutePerformanceTest(ctx context.Context) error {
+	template, err := tmpl.Parse(jobTemplate)
+	if err != nil {
+		return err
+	}
+
+	name := string(uuid.NewUUID())
+	t.templatedValues["HTMLReportPath"] = fmt.Sprintf("%s.html", name)
+
+	useExternalIngressGatewayHost := os.Getenv("USE_EXTERNAL_INGRESS_GATEWAY_HOST") == "true"
+	if useExternalIngressGatewayHost {
+		host := getRequestHost()
+		if host != "" {
```

Comment:
If the host is empty, the default value from the job.yaml template will be used
I don't see an issue here, with this you can also run the test calling the svc directly

There's a print for this in `getRequestHost()`

Reply:
OK, you might be right now that I think of it :D
The original reason behind this was to support calling the gRPC service directly, but then I changed the default address to `istio-ingressgateway.istio-system.svc.cluster.local:443`
It might be better than to change the default to `grpcbin:50051`, so the test would also support running it without istio